src/aind_metadata_mapper/U19/procedures.py

```diff
@@ -0,0 +1,409 @@
+"""Sets up the U19 ingest ETL"""
+
+import json
+import logging
+from datetime import datetime
+from pathlib import Path
+from typing import List, Optional, Union
+
+import pandas as pd
+import requests
+from aind_data_schema.core.procedures import (
+    Procedures,
+    Reagent,
+    SpecimenProcedure,
+    SpecimenProcedureType,
+)
+from aind_data_schema_models.organizations import Organization
+from pydantic import Field
+from pydantic_settings import BaseSettings
+
+from aind_metadata_mapper.core import GenericEtl, JobResponse
+from aind_metadata_mapper.U19.utils import construct_new_model
+
+
+class JobSettings(BaseSettings):
+    """Data that needs to be input by user."""
+
+    tissue_sheet_path: Path
+    tissue_sheet_names: List[str]
+    output_directory: Optional[Path] = Field(
+        default=None,
+        description=(
+            "Directory where to save the json file to. If None, then json"
+            " contents will be returned in the Response message."
+        ),
+    )
+    experimenter_full_name: List[str]
+    subject_to_ingest: str = Field(
+        default=None,
+        description=(
+            "subject ID to ingest. If None,"
+            " then all subjects in spreadsheet will be ingested."
+        ),
+    )
+    procedures_download_link: str = Field(
+        default="http://aind-metadata-service/procedures",
+        description="Link to download the relevant procedures "
+        "from metadata service",
+    )
+    allow_validation_errors: bool = Field(
+        False, description="Whether or not to allow validation errors."
+    )
+
+
+def get_dates(string):
+    """Get the dates from a string."""
+    return [str.strip(date) for date in string.split(" - ")]
+
+
+def strings_to_dates(strings):
```

Comment:
what are these 2 dates? Instead of returning a tuple it might be useful to separate these